### ISSUE TITLE
Two questions about weight matrix W and update rules for attribute h.

### ISSUE THREAD

**Issue by YongZh131** [marble0117/gcnns#2]
Excuse me, is the adjacency matrix A used instead of the weight matrix W in masked_gcn? In addition, is the update rule of attribute h in your code different from that in the paper?

**Comment by marblet** [comment]
I understand my implementation is not correct.
I apologize for the inconvenience.

**Comment by marblet** [comment]
Now I fixed the code of Masked-GCN, however, I could not reproduce the score in the paper.
Could you find the reason for this?
Here is the implementation link:
https://github.com/marblet/gcnns/blob/master/models/masked_gcn.py

**Comment by YongZh131** [comment]
> Now I fixed the code of Masked-GCN, however, I could not reproduce the score in the paper.
> Could you find the reason for this?
> Here is the implementation link:
> https://github.com/marblet/gcnns/blob/master/models/masked_gcn.py

I'm also confused about getting a lower score. So I tried to contact the authors. The following is the author's reply: I'm sorry, the code of this work is on the cluster of the lab, and it can't be accessed now. Try to send it to you after school starts.
As you know, most Chinese schools start around September 1st. Until then, we can continue to study the paper and reproduce its score. If the author replies to me again, I will notify you immediately.
In addition, the edge weight w_ij of Masked GCN (Sym) is equal to the element of the adjacency matrix A at position (i, j), that is, w_ij=a_ij. This is mentioned in Remark 3 in Section 4.2. But in your code, w_ij seems to be obtained by Eq. (1). I’m not sure if I’m right, what do you think?
In addition, I don't know what the objective functions such as Eq. (16) and Eq. (20) refer to. Are they equivalent to loss function or cost function? Or just used to derive feature update rules, i.e. Eq. (17)?

**Comment by marblet** [comment]
I am sorry for replying late.
I tried both Eq. (1) and w_ij = a_ij for Sym_Masked GCN, but I could not reproduce the score.
And as you said I am also not sure what is the correct loss function. I assumed that label softmax loss is the loss for Masked GCN.

BTW, Thanks for sending email to the author.
Maybe it is better to wait the official source codes from the author...
